h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -248,6 +249,12 @@ public void compute2() {
                           computeCrossValidation();
                           tryComplete();
                         }
+                        @Override
+                        public boolean onExceptionalCompletion(Throwable ex, CountedCompleter caller) {
+                          Log.warn("Model training job "+_job._description+" completed with exception: "+ex);
+                          if (_job._result != null) _job._result.remove(); //ensure there's no incomplete model left for manipulation after crash or cancellation
```

Comment:
If we are removing an inconsistent model - and we can expect that because we got an exception - what happens when remove itself throws an exception?

Reply:
^ Good point.

Reply:
at the time the job executes onExceptionalCompletion, the job is already considered as completed, so I believe the new exception will be propagated to the job barrier as if the job failed entirely, but I need to double-check this and will come back with definitive answer.

Reply:
was a very good point indeed, apparently it creates a deadlock: the jobs never completes because the exception fails to be propagating to the barrier: fix coming asap